src/utils/components/UserCredentials/useSSHCommand.ts

```diff
@@ -0,0 +1,46 @@
+import { InfrastructureModel, modelToGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getCloudInitCredentials } from '@kubevirt-utils/resources/vmi';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+export type useSSHCommandResult = {
+  command: string;
+  user: string;
+  sshServiceRunning: boolean;
+};
+
+const getInfrastructureAPIURL = (infrastructure: any): string =>
+  infrastructure && infrastructure?.status ? infrastructure?.status.apiServerURL : undefined;
+
+const useSSHCommand = (
+  vmi: V1VirtualMachineInstance,
+  sshService: IoK8sApiCoreV1Service,
+): useSSHCommandResult => {
+  const [infrastructure, infrastructureLoaded, infrastructureError] =
+    useK8sWatchResource<K8sResourceCommon>({
+      groupVersionKind: modelToGroupVersionKind(InfrastructureModel),
+      namespace: vmi?.metadata?.namespace,
+      name: 'cluster',
+      isList: false,
+    });
+
+  const infrastuctureApiUrl =
+    infrastructureLoaded && !infrastructureError && getInfrastructureAPIURL(infrastructure);
+  const apiHostname = infrastuctureApiUrl && new URL(infrastuctureApiUrl).hostname;
+  const consoleHostname = window.location.hostname; // fallback to console hostname
+
+  const { user } = getCloudInitCredentials(vmi);
+  const sshServicePort = sshService?.spec?.ports?.find(
```

Comment:
i think i saw this function in another file, so it can be exported to util

Reply:
Do you think utils/utils.ts can be a good file?

Reply:
yes can be :)